tutorials/notebooks/Object-Oriented/Object-Oriented.ipynb

```diff
@@ -0,0 +1,593 @@
+    "    def __init__(self, times=None, fluxes=None, uncertainties=None, flags=None, name=None):\n",
+    "        self.times = times\n",
+    "        self.fluxes = fluxes\n",
+    "        self.uncertainties = uncertainties\n",
+    "        self.flags = flags\n",
+    "        self.name = name\n",
+    "            \n",
+    "    def plot(self, color=None):\n",
+    "        \"\"\"Plot the light curve\"\"\"\n",
+    "        plt.errorbar(self.times, self.fluxes, self.uncertainties, fmt='o', color=color)\n",
+    "        plt.xlabel('Time')\n",
+    "        plt.ylabel('Flux')\n",
+    "        plt.title(self.name)\n",
+    "\n",
+    "# Initialize a LightCurve object:\n",
+    "prox_cen = LightCurve(times=times, fluxes=fluxes, \n",
+    "                      uncertainties=uncertainties, flags=flags,\n",
+    "                      name=name)\n",
+    "\n",
+    "prox_cen.plot('y')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that you can access the attributes of the object within methods by calling `self.<attribute name>`.\n",
+    "\n",
+    "### Class methods\n",
+    "\n",
+    "There's more than one way to initialize a light curve. Maybe your light curves come to you in a particular file type, and you want to be able to load those files directly into a light curve object. You could do that with a _class method_, like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class LightCurve(object):\n",
+    "    def __init__(self, times=None, fluxes=None, uncertainties=None, flags=None, name=None):\n",
+    "        self.times = times\n",
+    "        self.fluxes = fluxes\n",
+    "        self.uncertainties = uncertainties\n",
+    "        self.flags = flags\n",
+    "        self.name = name\n",
+    "            \n",
+    "    def plot(self):\n",
+    "        \"\"\"Plot the light curve\"\"\"\n",
+    "        plt.errorbar(self.times, self.fluxes, self.uncertainties, fmt='o')\n",
+    "        plt.xlabel('Time')\n",
+    "        plt.ylabel('Flux')\n",
+    "        plt.title(self.name)\n",
+    "\n",
+    "    @classmethod\n",
+    "    def from_txt(cls, path):\n",
+    "        data = np.loadtxt(path)\n",
+    "        \n",
+    "        times = data[0, :]\n",
+    "        fluxes = data[1, :]\n",
+    "        uncertainties = data[2, :]\n",
+    "        flags = data[3, :]\n",
+    "        return cls(times=times, fluxes=fluxes, \n",
+    "                   uncertainties=uncertainties, flags=flags)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To get sample light curves to load using the new class method, run the script `generate_lcs.py` from the command line, or with the cell below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile generate_lcs.py\n",
```

Comment:
Is the `%%writefile` magic useable in this website framework?

Reply:
This should work, but we might need to double check that it writes to the correct location